core/cautils/getter/kscloudapi.go

```diff
@@ -45,3 +50,25 @@ func GetKSCloudAPIConnector() *v1.KSCloudAPI {
 
 	return &client
 }
+
+// APIPostRequest provides a low-level utility that sends a POST request to a given url
+func APIPostRequest(fullURL string, body []byte, headers map[string]string) (io.ReadCloser, int64, error) {
```

Comment:
do you mind adding a test for that? ideally you would start an http listener for the test and verify the correct post and send back an answer

Reply:
Yeah, I just wanted to know if I was on the right path first. I'll go ahead and do that.